grails-app/services/org/pih/warehouse/core/IdentifierService.groovy

```diff
@@ -335,7 +386,7 @@ class IdentifierService {
         def products = Product.findAll("from Product as p where p.active = true and (p.productCode is null or p.productCode = '')")
         products.each { product ->
             try {
-                def productCode = generateProductIdentifier()
+                def productCode = product.productType ? generateProductIdentifier(product.productType) : generateProductIdentifier()
```

Comment:
Oh shit, good catch. This would likely happen a bunch if we imported products without product code.